meson.build

```diff
@@ -65,3 +65,5 @@ subdir('data')
 subdir('doc')
 subdir('profile.d')
 subdir('src')
+
+meson.add_install_script('meson_post_install.py')
```

Comment:
`meson.add_install_script()` already has `DESTDIR` argument.

Reply:
You mean [skip_if_destdir](https://mesonbuild.com/Reference-manual_builtin_meson.html#mesonadd_install_script) ?  I thought about using it, but then thought that if the script evolves more, then we might not be able to use it anymore.

Up to you.

Reply:
When using skip_if_destdir, meson internally detects that it doesn't need to fork and run the install script.

So you may still be interested in hoisting that check out of the script and into meson's installdata.

Either way it is possible to later on replace `meson.add_install_script('systemd-tmpfiles', skip_if_destdir: true)` with `meson.add_install_script('meson_post_install.py', skip_if_destdir: true)`

Reply:
It's no longer sufficient to just check for `DESTDIR`.  Now we need to look for the absence of `/run/.containerenv` as well.

(This is what I meant by *if the script evolves more, then we might not be able to use it anymore*.)

Reply:
`skip_if_destdir: true` can overlap and coexist with checking for `/run/.containerenv`.

```diff
diff --git a/meson.build b/meson.build
index cf32ae3..8c41fd6 100644
--- a/meson.build
+++ b/meson.build
@@ -71,4 +71,4 @@ subdir('doc')
 subdir('profile.d')
 subdir('src')
 
-meson.add_install_script('meson_post_install.py')
+meson.add_install_script('meson_post_install.py', skip_if_destdir: true)
diff --git a/meson_post_install.py b/meson_post_install.py
index 2e22d87..c845e2a 100644
--- a/meson_post_install.py
+++ b/meson_post_install.py
@@ -4,9 +4,7 @@ import os
 import subprocess
 import sys
 
-destdir = os.environ.get('DESTDIR', '')
-
-if not destdir and not os.path.exists('/run/.containerenv'):
+if not os.path.exists('/run/.containerenv'):
     print('Calling systemd-tmpfiles --create ...')
 
     try:
```

Reply:
I was considering using `skip_if_destdir` when the post-install script was essentially a one-liner, because that was the [original motivation](https://github.com/mesonbuild/meson/issues/9) behind adding that to Meson as an exception to the general rule about environment variables.

However, it's not a one-liner anymore because of the second condition.  So we might as well keep everything in one place.

Reply:
Your link doesn't defend your claim that skip_if_destdir is an exception to "don't use environment variables.

Meson avoids environment variables during `mesin setup` because e.g. they result in different ninja build files if the environment variable changes on the next reconfigure.

The DESTDIR variable is only read during install, and meson already reads it to set the default `--destdir=...` value.

The skip_if_destdir kwarg has three practical effects here:
- it is processed in a fraction of a second if DESTDIR is set, and simply notes that it is skipped, rather than running in several fractions of a second by forking to python and doing nothing
- it helpfully logs "skipping custom install script because DESTDIR is set" (your script instead logs "running custom install script" and does nothing)
- it doesn't set the "did_install_something" bit, which may actually matter if no other files are installed, for example due to the use of install tags, in which case meson will log "nothing to install".

Nothing really serious, but my point is that on general principle there are reasons why for consistency, scripts should be using the skip_if_destdir kwarg.

Reply:
> Your link doesn't defend your claim that skip_if_destdir is an exception to
> "don't use environment variables.

`skip_if_destdir` was added in https://github.com/mesonbuild/meson/pull/8269  It says:
```
It is common, at least in GNOME projects, to have scripts that must be
run only in the final destination, to update system icon cache, etc.
```

That's exactly what ptomato was saying [earlier](https://github.com/mesonbuild/meson/issues/9#issuecomment-385849310).

As a GNOME developer, I have dealt with my share of Meson post-install scripts to get the connection. ;)

> The skip_if_destdir kwarg has three practical effects here:

I don't see any of those reasons to be important enough to sacrifice the readability of the build scripts.  Without getting into too many details why they don't matter, I will just say that if `skip_if_destdir` was that important, then it wouldn't be (finally) added to Meson in January 2021.  ;)

Reply:
Okay? That is not providing access to environment variables, and ptomato's mentioning DESTDIR in that issue was technically offtopic. DESTDIR is handled via minstall, always has been and always will be, and meson.build will still not read environment variables at configure time. (If it did then you'd have install scripts that ran during a DESTDIR install because DESTDIR wasn't set during `meson setup`...)

> I don't see any of those reasons to be important enough to sacrifice the readability of the build scripts.

That's a reasonable argument to make :) yes.